### PR TITLE
Feature/login pages

### LINKED ISSUE #2: עיצוב קופסת קלט שתומכת ב-react-hook-form
בסך הכל קופסת קלט שאחר כך יהיה אפשר להשתמש בה בשביל פורמים לכניסה

### LINKED ISSUE #3: מימוש שני הפורמים של ההתחברות
כולל input validation בעזרת react-hook-form שהשם יהיה מינימום שני אותיות, שהאימייל יהיה כתובת אמיתית ושלסיסמה יהיו לפחות 8 אותיות, אות קטנה אחת, אות גדולה אחת, מספר אחד וסימן מיוחד אחד. בנוסף תהיה את האפשרות להראות ולהסתיר את הסיסמא

regex for valid email:
`/^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,4}$`
regex for valid password:
`/^(?=.*\d)(?=.*[a-z])(?=.*[A-Z])(?=.*[a-zA-Z]).{8,}$/`

כולל יצירת המשתמש/כניסת המשתמש בעזרת פיירבייס

Relies on: #2

### PR DESCRIPTION
**Pull request by RealByter**

### CODE REVIEW

**Review by nickBes — COMMENTED** [review]
Some things don't work for me, so I couldn't really review this PR.

1. The submit event didn't execute not in the Sign In page nor in the Sign Up page.
2. Didn't see any indications when writing incorrect input values.

I'm running windows 10. Removed `package-lock.json` and `node-modules`, reinstalled everything and then executed the firebase emulator and the client.

File: client/src/pages/SignUp.tsx [hunk 1]
```diff
@@ -1,28 +1,31 @@
...
 const SignUpPage = () => {
   const navigate = useNavigate();
-  const [createUserWithEmailAndPassword, user] = useCreateUserWithEmailAndPassword(auth);
+  const [createUserWithEmailAndPassword] = useCreateUserWithEmailAndPassword(auth);
+  const [updateProfile] = useUpdateProfile(auth);
 
-  const handleFormSubmit = ({ email, password }: FormOptions) => {
-    createUserWithEmailAndPassword(email as string, password as string);
+  const handleFormSubmit = async ({ name, email, password }: FormOptions) => {
+    console.log('here');
```
Comment (nickBes) [inline client/src/pages/SignUp.tsx]: Please remove the unnecessary log

File: client/src/components/Form.tsx [hunk 2]
```diff
@@ -17,120 +22,49 @@ type FormProps = {
...
-          )}
-        </div>
+  } = useForm<FormOptions>();
 
-        <button type="submit" className={classes['form-submit-btn']}>
-          {submitLabel}
-        </button>
+  return (
+    <div className="wrapper">
+      <form onSubmit={handleSubmit(onSubmit)} className="form">
```
Comment (nickBes) [inline client/src/components/Form.tsx:35]: I personally think that making a single form for different pages could raise some issues while making changes or additions. For example, the way you choose email and password validators based on whether we want a name input. Or, when passing submit events we need to check whether some of the fields exist.

But if you don't think this is critical, leave it as is

File: client/src/components/FormField.tsx [hunk 3]
```diff
@@ -1,46 +1,47 @@
...
+          <button type="button" onClick={() => setShowPassword((prev) => !prev)}>
+            {showPassword ? (
+              <VisibilityOffOutlinedIcon htmlColor="#304571" />
+            ) : (
+              <VisibilityOutlinedIcon htmlColor="#304571" />
+            )}
+          </button>
         )}
       </div>
+      {error && <p className='error' dir='rtl'>{error.message}</p>}
```
Comment (nickBes) [inline client/src/components/FormField.tsx:44]: I know it wasn't mentioned in the mockup, but the error messages are too close to the input labels.
You could maybe move it to the left or open a new issue

image

File: client/src/pages/SignUp.tsx [hunk 4]
```diff
@@ -1,28 +1,31 @@
...
+      if (success) {
+        navigate('/');
+      }
+    }
   };
 
-  if (user) {
-    navigate('/');
-  }
-
```
Comment (nickBes) [inline client/src/pages/SignUp.tsx]: This was used to redirect people who are already signed in.
This is also relevant to the sign in page.

File: client/src/pages/SignUp.tsx [hunk 5]
```diff
@@ -1,28 +1,31 @@
...
 
 const SignUpPage = () => {
   const navigate = useNavigate();
-  const [createUserWithEmailAndPassword, user] = useCreateUserWithEmailAndPassword(auth);
+  const [createUserWithEmailAndPassword] = useCreateUserWithEmailAndPassword(auth);
+  const [updateProfile] = useUpdateProfile(auth);
 
-  const handleFormSubmit = ({ email, password }: FormOptions) => {
-    createUserWithEmailAndPassword(email as string, password as string);
+  const handleFormSubmit = async ({ name, email, password }: FormOptions) => {
```
Comment (nickBes) [inline client/src/pages/SignUp.tsx:19]: Kind of weird that we didn't mention error handling during authorization, or in any other process.
Maybe we should open an issue about that, I think it's kind of critical for the end user.

**Review by nickBes — CHANGES_REQUESTED** [review]

File: client/src/components/FormField.tsx (hunk 3 above)
Comment (RealByter) [reply client/src/components/FormField.tsx:44]: רשמתי את זה למעצבת, אני אנסה להזכיר לה. בעיקרון הודעות השגיאה לא אמורות להזיז את ה-layout כך שהבעיה היא עם המרחקים שיש עכשיו

File: client/src/components/Form.tsx (hunk 2 above)
Comment (RealByter) [reply client/src/components/Form.tsx:35]: דיברתי עם יעל לגבי זה וזה מה שהיא ביקשה ממני לעשות

File: client/src/pages/SignUp.tsx (hunk 5 above)
Comment (RealByter) [reply client/src/pages/SignUp.tsx:19]: גם זה משהו שאמור להיות מטופל על ידי המעצבת. אין את זה כרגע ואנחנו צריכים את הפורמים כך שהמשכתי הלאה

**Review by nickBes — APPROVED** [review]
Looks good

**Pull request merged**